src/LocationService.Abstractions/ILocationService.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Device.Location;
+
+namespace LocationService
+{
+	/// <summary>
+	/// A service which handles the permission to get the user's location and tracks that location.
+	/// </summary>
+	public interface ILocationService
+	{
+		/// <summary>
+		/// Gets the current location. If the permission is not already granted, throws an InvalidOperationException.
+		/// </summary>
+		Task<GeoCoordinate> GetLocation(CancellationToken ct);
+
+		/// <summary>
+		/// Gets an event indicating that the user location has changed.
+		/// </summary>
+		event EventHandler<GeoCoordinate> LocationChanged;
+
+		/// <summary>
+		/// Requests the location permission from the user. May do nothing if <see cref="IsPermissionRequested"/> is true.
+		/// </summary>
+		Task<bool> RequestPermission(CancellationToken ct);
+
+		/// <summary>
+		/// Gets whether the user has granted the location permission.
```

Comment:
This will never prompt the user right? If so we could add it to the summary for extreme clarity :)